app/actions/service_credential_binding_create.rb

```diff
@@ -0,0 +1,68 @@
+module VCAP::CloudController
+  module V3
+    class ServiceCredentialBindingCreate
+      class UnprocessableCreate < StandardError
+      end
+
+      def initialize(user_audit_info)
+        @user_audit_info = user_audit_info
+      end
+
+      def precursor(service_instance, app: nil, name: nil)
+        not_supported! unless service_instance.user_provided_instance?
+        app_is_required! unless app.present?
+
+        binding_details = {
+          service_instance: service_instance,
+          name: name,
+          app: app,
+          type: 'app',
+          credentials: service_instance.credentials,
+          syslog_drain_url: service_instance.syslog_drain_url
+        }
+
+        ServiceBinding.new(**binding_details).tap do |b|
+          b.save_with_new_operation(
+            {
+              type: 'create',
+              state: 'in progress',
+            }
+          )
+        end
```

Comment:
I am a bit confused as to why this is doing the binding and also setting "create in progress" when it is already storing the details of the binding (cred and sys drain log). 

Would it be better to not store the cred and sys log drain here, but use the client.bind as part of the `bind` (where client is` VCAP::Services::ServiceBrokers::UserProvided::Client`)? 
That way `bind` is doing the actual binding and the client knows what properties are needed for a bind

Reply:
It depends what we understand as a binding. One could say that as soon as that there's a row in the service_bindings table we have a binding. I see the point of maybe storing the creds only at `bind`. I didn't feel the need to add the ServiceBrokers::UserProvided::Client as a dependency (I actually didn't know it existed). I'll take a look at it.